JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -468,81 +400,92 @@ private fun InterestsTabRowContent(
-            if (windowMaxWidth < 800.dp) min(itemMaxWidth, 350.dp) else min(itemMaxWidth, 450.dp)
+            val maxWidthWithSpaces = outerConstraints.maxWidth - (columns - 1) * itemSpacingPx
+            (maxWidthWithSpaces / columns).coerceIn(0, itemMaxWidthPx)
         }
-    }
 
-@Composable
-fun rememberTabContent(
-    windowSize: WindowSize,
-    interestsViewModel: InterestsViewModel
-): List<TabContent> {
-    // UiState of the InterestsScreen
-    val uiState by interestsViewModel.uiState.collectAsState()
-
-    // Describe the screen sections here since each section needs 2 states and 1 event.
-    // Pass them to the stateless InterestsScreen using a tabContent.
-    val topicsSection = TabContent(Sections.Topics) {
-        val selectedTopics by interestsViewModel.selectedTopics.collectAsState()
-        TopicList(
-            windowSize = windowSize,
-            topics = uiState.topics,
-            selectedTopics = selectedTopics,
-            onTopicSelect = { interestsViewModel.toggleTopicSelection(it) }
-        )
-    }
+        // Keep track of the height of each row to calculate the layout's final size
+        val rowHeights = IntArray(measurables.size / columns + 1)
+        // Measure elements with their maximum width and keep track of the height
+        val placeables = measurables.mapIndexed { index, measureable ->
+            val itemConstraints = if (itemWidth == Int.MAX_VALUE) {
+                outerConstraints
+            } else {
+                outerConstraints.copy(maxWidth = itemWidth)
+            }
+            val placeable = measureable.measure(itemConstraints)
+            // Update the height for each row
+            val row = index.floorDiv(columns)
+            rowHeights[row] = max(rowHeights[row], placeable.height)
+            placeable
+        }
 
-    val peopleSection = TabContent(Sections.People) {
-        val selectedPeople by interestsViewModel.selectedPeople.collectAsState()
-        PeopleList(
-            people = uiState.people,
-            selectedPeople = selectedPeople,
-            onPersonSelect = { interestsViewModel.togglePersonSelected(it) }
-        )
-    }
+        // Calculate maxHeight of the Interests layout. Heights of the row + top padding
+        val layoutHeight = topPaddingPx + rowHeights.sumOf { it }
+        // Calculate maxWidth of the Interests layout
+        val layoutWidth = itemWidth * columns + (itemSpacingPx * (columns - 1))
+        // Lay out given the max width and height
+        layout(
+            outerConstraints.maxWidth.coerceIn(0, layoutWidth),
+            outerConstraints.maxHeight.coerceIn(0, layoutHeight)
+        ) {
+            // Track the y co-ord we have placed children up to
+            var yPosition = topPaddingPx
+            placeables.forEachIndexed { index, placeable ->
+                // If there's only one column, then place items one after the other
+                if (columns == 1) {
```

Comment:
If I'm reading this correctly, I think you don't have to have the special case for one column, since `index % 1` will always equal 0.

Reply:
Simplified with the change above